lib/evaluation/evaluationClientWrapper.ts

```diff
@@ -202,23 +202,39 @@ export class EvaluationClientWrapper {
   ): SubmissionRequirementMatch[] {
     const submissionRequirementMatches: SubmissionRequirementMatch[] = [];
     for (const sr of submissionRequirements) {
+      // Create a default SubmissionRequirementMatch object
+      const srm: SubmissionRequirementMatch = {
+        name: pd.name || pd.id,
+        rule: sr.rule,
+        vc_path: [],
+      };
+      if (sr.from) {
+        srm.from = sr.from;
+      }
+      // Assign min, max, and count regardless of 'from' or 'from_nested'
+      sr.min ? (srm.min = sr.min) : undefined;
+      sr.max ? (srm.max = sr.max) : undefined;
+      sr.count ? (srm.count = sr.count) : undefined;
+
       if (sr.from) {
         const matchingDescriptors = this.mapMatchingDescriptors(pd, sr, marked);
         if (matchingDescriptors) {
-          sr.min ? (matchingDescriptors.min = sr.min) : undefined;
-          sr.max ? (matchingDescriptors.max = sr.max) : undefined;
-          sr.count ? (matchingDescriptors.count = sr.count) : undefined;
-          submissionRequirementMatches.push(matchingDescriptors);
+          srm.vc_path.push(...matchingDescriptors.vc_path);
+          srm.name = matchingDescriptors.name;
+          submissionRequirementMatches.push(srm);
         }
       } else if (sr.from_nested) {
-        const srm: SubmissionRequirementMatch = { name: pd.name || pd.id, rule: sr.rule, from_nested: [], vc_path: [] };
-        if (srm && srm.from_nested) {
-          sr.min ? (srm.min = sr.min) : undefined;
-          sr.max ? (srm.max = sr.max) : undefined;
-          sr.count ? (srm.count = sr.count) : undefined;
-          srm.from_nested.push(...this.matchSubmissionRequirements(pd, sr.from_nested, marked));
+        // Recursive call to matchSubmissionRequirements for nested requirements
+        try {
+          srm.from_nested = this.matchSubmissionRequirements(pd, sr.from_nested, marked);
           submissionRequirementMatches.push(srm);
+        } catch (err) {
+          // Handle any errors that occur in the recursive call
```

Comment:
Why is the error discarded with a console.log in the end?

Reply:
changed. I'm throwing an error instead and handling it there